src/goose.rs

```diff
@@ -1421,14 +1422,13 @@ impl GooseClient {
     /// }
     /// ```
     pub async fn set_client_builder(&self, builder: ClientBuilder) {
-        let client = match builder.build() {
-            Ok(c) => Mutex::new(c),
+        match builder.build() {
+            Ok(c) => *self.client.lock().await = c,
```

Comment:
Can we use a mutable helper variable here? - this syntax while correct looks a little bit weird.

Reply:
You mean something like this? I personally prefer the existing code.

```
diff --git a/src/goose.rs b/src/goose.rs
index aa497a1..9693a2d 100644
--- a/src/goose.rs
+++ b/src/goose.rs
@@ -1423,7 +1423,10 @@ impl GooseClient {
     /// ```
     pub async fn set_client_builder(&self, builder: ClientBuilder) {
         match builder.build() {
-            Ok(c) => *self.client.lock().await = c,
+            Ok(c) => {
+                let mut client = self.client.lock().await;
+                *client = c;
+            },
             Err(e) => {
                 error!("failed to build client: {}", e);
                 std::process::exit(1);
```